src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/EditorContextScope.cs

```diff
@@ -59,7 +59,11 @@ public void AddPostCheckExpressionRewriter(IExpressionRewriter expressionRewrite
 
         // Host can set optional hints about where this expression is used. 
         // This can feed into Fx2NL and other help services. 
-        public UsageHints UsageHints { get; init; }
+        public UsageHints UsageHints { get; init; }
+
+        // Host can set optional range of expression.
+        // This can feed into Fx2NL and other help services. 
+        public ScalarRange Range { get; init; }
```

Comment:
can we avoid this? Nothing else in this class is specific to Fx2nl

Reply:
Sure I can remove this from here if not needed

Reply:
yes you don't even need this range field here. The UsageHints in fx2nlparameters was supposed to be provided from host's server to lsp but the range comes directly host's client and not the server which means it is already a part of customfx2nl params which you can just copy over to fx2nlparameters to transfer to nl library. Sounds awkward but that's more convenient for hosts